agoric/contract/test/swingsetTests/bootstrap/bootstrap-mint.js

```diff
@@ -0,0 +1,581 @@
+    );
+    throw error;
+  }
+}
+
+export async function mintNoCharactersAvailable(context) {
+  /** @type {Context} */
+  const {
+    publicFacet,
+    paymentAsset,
+    users: { alice },
+    zoe,
+  } = context;
+  const { offerArgs, message, give } = flow.mintCharacter.noAvailability;
+  
+  const mintCharacterInvitation = await E(
+    publicFacet,
+  ).makeMintCharacterInvitation();
+  const priceAmount = AmountMath.make(paymentAsset.brandMockIST, give.Price);
+
+  const proposal = harden({
+    give: { Price: priceAmount },
+  });
+
+  const payment = { Price: await alice.withdrawPayment(priceAmount) };
+
+  const userSeat = await E(zoe).offer(
+    mintCharacterInvitation,
+    proposal,
+    payment,
+    offerArgs,
+  );
+
+  try {
+    await E(userSeat).getOfferResult();
+  } catch (error) {
+    assert.equal(error.message, message);
+
+    const characters = await E(publicFacet).getCharacters();
+    assert.equal(
+      characters.length,
+      1,
+      'New character was not added to contract registry due to mint error',
+    );
+    throw error;
+  }
+}
+
+export async function mintInventoryCheck(context) {
+  /** @type {Context} */
+  const { publicFacet } = context;
+  const { offerArgs } = flow.mintCharacter.expected;
+
+  const characterInventory = await E(publicFacet).getCharacterInventory(
+    offerArgs.name,
+  );
+
+  const mappedInventory = characterInventory.items.map((i) => i[0]);
+
+  assert.equal(
+    mappedInventory.length,
+    3,
+    'New character inventory does not contain 3 items',
+  );
+
+  assert.equal(
+    new Set(mappedInventory.map((i) => i.category)).size,
+    3,
+    'Two or more items have the same category',
+  );
+
+  assert.equal(
+    mappedInventory.filter((i) => i.rarity < 20).length,
+    2,
+    'No two common items',
+  );
+
+  assert.equal(
+    mappedInventory.filter((i) => i.rarity > 19).length,
+    1,
+    'No uncommon to legendary item found.',
+  );
+}
+
+export async function mintItemExpectedFlow(context) {
+  /** @type {Context} */
+  const { creatorFacet, contractAssets, purses, zoe } = context;
+  const { want, message } = flow.mintItem.expected;
+
+  const mintItemInvitation = await E(creatorFacet).makeMintItemInvitation();
+  const proposal = harden({
+    want: {
+      Item: AmountMath.make(
+        contractAssets.item.brand,
+        makeCopyBag(harden([[want, 1n]])),
+      ),
+    },
+  });
+
+  const userSeat = await E(zoe).offer(mintItemInvitation, proposal);
+
+  const result = await E(userSeat).getOfferResult();
+  assert.equal(result, message, 'Offer does not return success message');
+
+  const payout = await E(userSeat).getPayout('Asset');
+  await E(purses.item).deposit(payout);
+  assert.equal(
+    (await E(purses.item).getCurrentAmount()).value.payload[0][0].name,
+    want.name,
+    'New Item was not added to character purse',
+  );
+}
+
+export async function mintSameItemSFT(context) {
```

Comment:
I think all the assert messages here are backwards, minting the same item should and does succeed. Let's update the text to reflect that

Reply:
Yeah I changed all the messages around because the message is logged when the assertion fails. I think we implemented it wrong beforehand. So here:
`assert.equal(result, message, 'Offer does not return success message');` (from the item mint function you commented on)
where message is `'Item NFT(s) minted successfully!'`

When this assertions fails, it means that the offer result returned is not `'Item NFT(s) minted successfully!'` hence we want to log to the console something like offer does not return success message.